docs/types/private.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Private Fields
+---
+
+# Private Fields
+
+Private (external) fields can provide local context for later resulution.
+These fields will act as plain dataclass fields and not become attached to
+strawberry resolvers.
+
+Some uses include:
+
+- Context that relies upon field inputs.
+- Avoiding fully materializing an object hierarchy (lazy resolution)
+
+# Defining a private field
+
+Specifying a field with `strawberry.Private[...]` will desigate the field
+as for dataclasses, and not a graphql field.
```

Comment:
I wouldn't use dataclasses terminology here, we shouldn't assume people know what dataclasses are or how they work 😊

Reply:
yes, that might be better 😊